src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -12,12 +12,13 @@ import { samLambdaRuntimes } from '../../../lambda/models/samLambdaRuntime'
 import { CloudFormation } from '../../cloudformation/cloudformation'
 import { CloudFormationTemplateRegistry } from '../../cloudformation/templateRegistry'
 import { isContainedWithinDirectory } from '../../filesystemUtilities'
-import { AwsSamDebuggerConfiguration, AwsSamDebuggerInvokeTargetTemplateFields } from './awsSamDebugConfiguration'
+import { AwsSamDebuggerInvokeTargetTemplateFields } from './awsSamDebugConfiguration'
+import { AwsSamDebuggerConfiguration } from './awsSamDebugConfiguration.gen'
 
 export const AWS_SAM_DEBUG_TYPE = 'aws-sam'
 export const DIRECT_INVOKE_TYPE = 'direct-invoke'
-export const TEMPLATE_TARGET_TYPE = 'template'
-export const CODE_TARGET_TYPE = 'code'
+export const TEMPLATE_TARGET_TYPE: 'template' = 'template'
```

Comment:
because the type has an enum, it gets generated as a string literal (e.g. 'code' | 'template') rather than a string - this means that when we declare variables we have to explicitly type them - else the type-checker blows up.

Better way?

Reply:
This is the enforcement I wanted to do but was unsure if I could do based on the enforcement that package.json gave us. What happens if you create a debug config with a bad target type? Will it even let you attempt to debug? If that's the case, then this is excellent (you might be able to swap it to something like `export type TEMPLATE_TARGET_TYPE = 'template'`).

Otherwise, I'd just let it accept a string wholesale in the interface and filter it out on the code level (as we did previously)

Reply:
I'm thinking that the code generation could assist here if that inlined enum is factored out into the `definitions` section.

Reply:
`export type TEMPLATE_TARGET_TYPE = 'template'` complains that it's a `type` not a `value` at the call-site. I googled a bit last night before doing the initial implementation and that's what led me to the above syntax, it's basically saying it's an *instance* of a variable whose *type* is `template` and whose *value* is `template`.